pkg/operator/ceph/file/mds/probes_test.go

```diff
@@ -0,0 +1,188 @@
+			"CEPH_MDS_DUMP_MOCK_RETCODE=0",
+			"JQ=" + jqPath,
+		}
+		stdout := bytes.NewBuffer([]byte{})
+		stderr := bytes.NewBuffer([]byte{})
+		cmd.Stdout = stdout
+		cmd.Stderr = stderr
+
+		err := cmd.Run()
+		assert.NoError(t, err)
+		assert.Equal(t, 0, cmd.ProcessState.ExitCode())
+		if t.Failed() {
+			t.Log("stdout:", stdout.String())
+			t.Log("stderr:", stderr.String())
+		}
+	})
+	t.Run("mds not in map", func(t *testing.T) {
+		cmd := exec.Command("bash", "-c", probeShimWrapper)
+		cmd.Env = []string{
+			// "ROOK_CEPH_MON_HOST" intentionally empty
+			// "ROOK_CEPH_MON_INITIAL_MEMBERS" intentionally empty
+			`CEPH_MDS_DUMP_MOCK_STDOUT={"active": "mds.b"}`, // mds.b != mds.a should result in failure
+			// "CEPH_MDS_DUMP_MOCK_STDERR" intentionally empty
+			"CEPH_MDS_DUMP_MOCK_RETCODE=0",
+			"JQ=" + jqPath,
+		}
+		stdout := bytes.NewBuffer([]byte{})
+		stderr := bytes.NewBuffer([]byte{})
+		cmd.Stdout = stdout
+		cmd.Stderr = stderr
+
+		err := cmd.Run()
+		assert.Error(t, err)
+		assert.Equal(t, 1, cmd.ProcessState.ExitCode())
+		if t.Failed() {
+			t.Log("stdout:", stdout.String())
+			t.Log("stderr:", stderr.String())
+		}
+	})
+	t.Run("ceph mds dump error", func(t *testing.T) {
+		cmd := exec.Command("bash", "-c", probeShimWrapper)
+		cmd.Env = []string{
+			// "ROOK_CEPH_MON_HOST" intentionally empty
+			// "ROOK_CEPH_MON_INITIAL_MEMBERS" intentionally empty
+			// "CEPH_MDS_DUMP_MOCK_STDOUT" intentionally empty
+			"CEPH_MDS_DUMP_MOCK_STDERR='mock failure'",
+			"CEPH_MDS_DUMP_MOCK_RETCODE=1",
+			"JQ=" + jqPath,
+		}
+		stdout := bytes.NewBuffer([]byte{})
+		stderr := bytes.NewBuffer([]byte{})
+		cmd.Stdout = stdout
+		cmd.Stderr = stderr
+
+		err := cmd.Run()
+		assert.NoError(t, err)
+		assert.Equal(t, 0, cmd.ProcessState.ExitCode())
+		if t.Failed() {
+			t.Log("stdout:", stdout.String())
+			t.Log("stderr:", stderr.String())
+		}
+	})
+	t.Run("ceph mds dump invalid json", func(t *testing.T) {
+		cmd := exec.Command("bash", "-c", probeShimWrapper)
+		cmd.Env = []string{
+			// "ROOK_CEPH_MON_HOST" intentionally empty
+			// "ROOK_CEPH_MON_INITIAL_MEMBERS" intentionally empty
+			`CEPH_MDS_DUMP_MOCK_STDOUT={"active": "mds.a"`, // missing end brace
+			// "CEPH_MDS_DUMP_MOCK_STDERR" intentionally empty
+			"CEPH_MDS_DUMP_MOCK_RETCODE=0",
+			"JQ=" + jqPath,
+		}
+		stdout := bytes.NewBuffer([]byte{})
+		stderr := bytes.NewBuffer([]byte{})
+		cmd.Stdout = stdout
+		cmd.Stderr = stderr
+
+		err := cmd.Run()
+		assert.NoError(t, err)
+		assert.Equal(t, 0, cmd.ProcessState.ExitCode())
+		if t.Failed() {
+			t.Log("stdout:", stdout.String())
+			t.Log("stderr:", stderr.String())
+		}
+	})
+	t.Run("ceph mds in standby list", func(t *testing.T) {
+		cmd := exec.Command("bash", "-c", probeShimWrapper)
+		cmd.Env = []string{
+			// "ROOK_CEPH_MON_HOST" intentionally empty
+			// "ROOK_CEPH_MON_INITIAL_MEMBERS" intentionally empty
+			`CEPH_MDS_DUMP_MOCK_STDOUT={"standby": "mds.a"}`, // standby mds.a == mds.a should result in success
```

Comment:
There need to be tests that check that the script behaves as expected with different outputs from `ceph ...` command(s). I tried to clarify that the place to do that is below based on the testing framework I suggested.

```suggestion
			`CEPH_MDS_DUMP_MOCK_STDOUT={<!-- replace with real output from ceph command -->}`, // standby mds.a == mds.a should result in success
```

Reply:
Updated, check the initial flow